packages/headless/src/state/product-listing-app-state.ts

```diff
@@ -0,0 +1,13 @@
+import {
+  ConfigurationSection,
+  ContextSection,
+  ProductRecommendationsSection,
+  SearchHubSection,
+  VersionSection,
+} from './state-sections';
+
+export type ProductListingAppState = ConfigurationSection &
```

Comment:
Is the query pipeline concept going to be leveraged in the new commerce product listing API ? Or is it going to be expressed in a different manner ? 

I guess this goes outside the pure scope of this PR, I'm mostly curious how the queries are going to be "routed" in Coveo :)

Reply:
Great question :)

They are routed through the *url* parameter :exploding_head: 

So you configure a product listing with a specific URL, say: `mysite.com/supersite/supersale`

When you `POST {url: 'mysite.com/supersite/supersale'}`, you will get the results that were configured in the admin console for this URL, which makes it very easy for a sales manager to select which results they want to show.

So in effect:

* No `pipeline` parameter! We want it to go through the process as-is. If you want to customize a pipeline for it, you will be able to add a condition on a pipeline like `isProductListing: true`.
* No searchHub, we don't know _exactly_ how it will take form, but we for sure want to do it server-side
* No analytics, we also want to do it server-side

So it should make for a pretty lean API :smile:

Reply:
Seems awesome ! Thanks for the answer